torch/csrc/autograd/functions/batch_normalization.cpp

```diff
@@ -206,7 +206,7 @@ auto BatchNormBackward::apply(const variable_list& grad_outputs) -> variable_lis
                                  std::move(grad_bias));
   return wrap_outputs(all_inputs, std::move(outputs), [&](FunctionFlags f) {
     return std::make_shared<BatchNormBackwardBackward>(
-      f, *this, std::move(save_mean), std::move(save_std),
```

Comment:
just for my own understanding, what makes this non-reentrant, and why is saving references to save_mean / save_std no longer needed?